utils/testserver/index.ts

```diff
@@ -14,65 +15,67 @@
+import { join } from 'path';
+import { promisify } from 'util';
+import { Server as WebSocketServer } from 'ws';
+import zlib, { gzip } from 'zlib';
 
 const fulfillSymbol = Symbol('fullfil callback');
 const rejectSymbol = Symbol('reject callback');
 
-const gzipAsync = util.promisify(zlib.gzip.bind(zlib));
+const gzipAsync = promisify(gzip.bind(zlib));
 
-class TestServer {
-  /**
-   * @param {string} dirPath
-   * @param {number} port
-   * @param {string=} loopback
-   * @return {!Promise<TestServer>}
-   */
-  static async create(dirPath, port, loopback) {
+export class TestServer {
+  private _server: http.Server;
+  private _wsServer: WebSocketServer;
+  private _dirPath: string;
+  readonly debugServer: any;
+  private _startTime: Date;
+  private _cachedPathPrefix: string | null;
+  private _sockets = new Set<Socket>();
+  private _routes = new Map<string, (arg0: http.IncomingMessage, arg1: http.ServerResponse) => any>();
+  private _auths = new Map<string, { username: string; password: string; }>();
+  private _csp = new Map<string, string>();
+  private _extraHeaders = new Map<string, object>();
+  private _gzipRoutes = new Set<string>();
+  private _requestSubscribers = new Map<string, Promise<any>>();
+  readonly PORT: number;
+  readonly PREFIX: string;
+  readonly CROSS_PROCESS_PREFIX: string;
+  readonly EMPTY_PAGE: string;
+
+  static async create(dirPath: string, port: number, loopback?: string): Promise<TestServer> {
     const server = new TestServer(dirPath, port, loopback);
     await new Promise(x => server._server.once('listening', x));
     return server;
   }
 
-  /**
-   * @param {string} dirPath
-   * @param {number} port
-   * @param {string=} loopback
-   * @return {!Promise<TestServer>}
-   */
-  static async createHTTPS(dirPath, port, loopback) {
+  static async createHTTPS(dirPath: string, port: number, loopback?: string): Promise<TestServer> {
     const server = new TestServer(dirPath, port, loopback, {
-      key: await fs.promises.readFile(path.join(__dirname, 'key.pem')),
-      cert: await fs.promises.readFile(path.join(__dirname, 'cert.pem')),
+      key: await fs.promises.readFile(join(__dirname, 'key.pem')),
+      cert: await fs.promises.readFile(join(__dirname, 'cert.pem')),
       passphrase: 'aaaa',
     });
     await new Promise(x => server._server.once('listening', x));
     return server;
   }
 
-  /**
-   * @param {string} dirPath
-   * @param {number} port
-   * @param {string=} loopback
-   * @param {!Object=} sslOptions
-   */
-  constructor(dirPath, port, loopback, sslOptions) {
+  constructor(dirPath: string, port: number, loopback?: string, sslOptions?: object) {
     if (sslOptions)
-      this._server = https.createServer(sslOptions, this._onRequest.bind(this));
+      this._server = _createServer(sslOptions, this._onRequest.bind(this));
     else
-      this._server = http.createServer(this._onRequest.bind(this));
+      this._server = createServer(this._onRequest.bind(this));
```

Comment:
`http.createServer` would be much more readable :)

Reply:
You dont say!